inst/report/scripts/run_molevolvr_pipeline.R

```diff
@@ -0,0 +1,1296 @@
+# Author(s): Awa Synthia
+# Last modified: 2024
+
+# Load necessary libraries
+library(httr)
+library(data.table)
+library(readr)
+library(rentrez)
+
+get_sequences <- function(sequences,
+                          acc_file_path = "accs.txt",
+                          dir_path = "~",
+                          separate = TRUE) {
+  seqs <- readAAStringSet(sequences)
+  cln_names <- c()
+  for (accnum in names(seqs)) {
+    if (grepl("\\|", accnum)) {
+      accnum_cln <- strsplit(accnum, "\\|")[[1]][2]
+      accnum_cln <- strsplit(accnum_cln, " ")[[1]]
+    } else {
+      accnum_cln <- strsplit(accnum, " ")[[1]][1]
+    }
+    cln_names <- append(cln_names, accnum_cln)
+    write(accnum_cln, file = acc_file_path, append = TRUE)
+    if (separate) {
+      write(paste0(dir_path, "/", accnum_cln, ".faa"),
+            file = "input.txt", append = TRUE)
+      write(paste0(">", accnum_cln),
+            file = paste0(accnum_cln, ".faa"), append = TRUE)
+      write(toString(seqs[accnum]),
+            file = paste0(accnum_cln, ".faa"), append = TRUE)
+    }
+  }
+  names(seqs) <- cln_names
+  writeXStringSet(seqs, sequences, format = "fasta")
+  return(length(seqs))
+}
+submit_full <- function(
+    dir = "/data/scratch",
+    DB = Sys.getenv("BLAST_DB", unset = "refseq"),
+    NHITS = Sys.getenv("BLAST_HITS", unset = 100),
+    EVAL = Sys.getenv("BLAST_EVALUE", unset = 0.00001),
+    sequences = "~/test.fa",
+    phylo = "FALSE",
+    by_domain = "FALSE",
+    domain_starting = "~/domain_seqs.fa",
+    type = "full",
+    job_code=NULL,
+    submitter_email=NULL,
+    advanced_options=NULL,
+    get_slurm_mails=FALSE
+) {
+  # Set working directory
+  setwd(dir)
```

Comment:
@the-mayer, check to see if this is risky, or relative paths or the use of `here` might be better.